src/sentry/interfaces/contexts.py

```diff
@@ -0,0 +1,111 @@
+"""
+sentry.interfaces.contexts
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:copyright: (c) 2010-2014 by the Sentry Team, see AUTHORS for more details.
+:license: BSD, see LICENSE for more details.
+"""
+
+from __future__ import absolute_import
+
+from sentry.utils.safe import trim
+from sentry.interfaces.base import Interface
+
+
+__all__ = ('Contexts',)
+
+
+context_types = {}
+
+
+def contexttype(name):
+    def decorator(cls):
+        cls.type = name
+        context_types[name] = cls
+        return cls
+    return decorator
+
+
+class ContextType(object):
+    indexed_fields = []
+
+    def __init__(self, alias, data):
+        self.alias = alias
+        self.data = data
+
+    def to_json(self):
+        return self.data
+
+    def flatten_index_value(self, value):
```

Comment:
whats this mean? might be worthwhile to add some docstrings on the base type to explain input/output